helix-term/src/ui/editor.rs

```diff
@@ -651,6 +655,46 @@ impl EditorView {
         );
     }
 
+    /// Apply the highlighting on the lines where a cursor is active
+    pub fn highlight_cursorline(doc: &Document, view: &View, surface: &mut Surface, theme: &Theme) {
+        let text = doc.text().slice(..);
+        let last_line = view.last_line(doc);
+
+        let primary_line = doc.selection(view.id).primary().cursor_line(text);
+
+        // The secondary_lines do contain the primary_line, it doesn't matter
+        // as the else-if clause in the loop later won't test for the
+        // secondary_lines if primary_line == line.
+        // It's used inside a loop so the collect isn't needless:
+        // https://github.com/rust-lang/rust-clippy/issues/6164
+        #[allow(clippy::needless_collect)]
+        let secondary_lines: Vec<_> = doc
+            .selection(view.id)
+            .iter()
+            .map(|range| range.cursor_line(text))
+            .collect();
+
+        for line in view.offset.row..(last_line + 1) {
+            let area = Rect::new(
+                view.area.x,
+                view.area.y + (line - view.offset.row) as u16,
+                view.area.width,
+                1,
+            );
+            if primary_line == line {
+                let style = theme
+                    .try_get("ui.cursorline.primary")
+                    .unwrap_or_else(|| theme.get("ui.cursorline"));
```

Comment:
This can be simplified to `theme.get("ui.cursorline.primary")` and same for the secondary style below. `theme.get` falls back to ancestor scopes.

These can also both be moved outside the `for` loop.